packages/@aws-cdk/aws-eks/README.md

```diff
@@ -811,6 +812,125 @@ const chart2 = cluster.addHelmChart(...);
 chart2.node.addDependency(chart1);
 ```
 
+#### CDK8s Charts
+
+[CDK8s](https://cdk8s.io/) is an open-source library that enables Kubernetes manifest authoring using familiar programming languages. It is founded on the same technologies as the AWS CDK, such as [`constructs`](https://github.com/aws/constructs) and [`jsii`](https://github.com/aws/jsii).
+
+> To learn more about cdk8s, visit the [Getting Started](https://github.com/awslabs/cdk8s/tree/master/docs/getting-started) tutorials.
+
+The EKS module natively integrates with cdk8s and allows you to apply cdk8s charts on AWS EKS clusters via the `cluster.addCdk8sChart` method.
+
+In addition to `cdk8s`, you can also use [`cdk8s+`](https://github.com/awslabs/cdk8s/tree/master/packages/cdk8s-plus), which provides higher level abstraction for the core kubernetes api objects.
+You can think of it like the `L2` constructs for Kubernetes.
+
+To get started, add the following dependencies to your `package.json` file:
+
+```json
+dependencies: {
+  "cdk8s": "0.30.0",
+  "cdk8s-plus": "0.30.0",
+  "constructs": "3.0.4",
+}
+```
+
+> Note that the version of `cdk8s` must be `>=0.30.0`.
+
+We recommend to seperate the `cdk8s` charts to a different file and extend the `cdk8s.Chart` class.
+You can use `aws-cdk` construct attributes and properties inside your `cdk8s` construct freely.
+
+In this example we create a chart that accepts an `s3.Bucket` and passes its name to a kubernetes pod as an environment variable.
+
+```ts
+import * as s3 from '@aws-cdk/aws-s3';
+import * as constructs from 'constructs';
+import * as cdk8s from 'cdk8s';
+import * as kplus from 'cdk8s-plus';
+
+export interface MyChartProps {
+
+  readonly bucket: s3.Bucket;
+}
+
+export class MyChart extends cdk8s.Chart {
+  constructor(scope: constructs.Construct, id: string, props: MyChartProps} ) {
+    super(scope, id);
+
+    new kplus.Pod(this, 'Pod', {
+      spec: {
+        containers: [
+          new kplus.Container({
+            image: 'my-image',
+            env: {
+              BUCKET_NAME: bucket.bucketName,
+            }
+          })
+        ]
+      }
+    });
+
+  }
+}
+```
+
+Then, in your cdk app:
+
+```ts
+import * as s3 from '@aws-cdk/aws-s3';
+import * as cdk8s from 'cdk8s';
+import { MyChart } from './my-chart'
+
+// some bucket..
+const bucket = new s3.Bucket(this, 'Bucket');
+
+// create a cdk8s chart and use `cdk8s.App` as the scope.
+const myChart = new MyChart(new cdk8s.App(), 'MyChart', { bucket });
+
+// add the cdk8s chart to the cluster
+cluster.addCdk8sChart('my-chart', myChart);
+```
+
+##### Custom CDK8s Constructs
```

Comment:
This section is pure `cdk8s`, the reason I embedded a concrete example instead of referring to `cdk8s` README is to emphasize that users must use the `constructs.Construct` class, which might be a bit un-natural inside a CDK application.